pkg/util/version/version.go

```diff
@@ -232,3 +255,41 @@ func (v *Version) Compare(w *Version) int {
 func (v *Version) AtLeast(w *Version) bool {
 	return v.Compare(w) >= 0
 }
+
+// UpgradeType returns the version change type that will happen when moving
+// between two version of CockroachDB.
+func (v *Version) UpgradeType(to *Version) UpgradeType {
```

Comment:
Given the only usage of this is to power the crdb_internal helper (see previous comment, I'm not sure if that's strictly the best way to interface with the stuff), I'd prefer if this logic stayed outside of the version package and was just in that crdb_internal method. It would also get rid of all these extra UpgradeType symbols in the otherwise tiny package. 

---

Actually, hm, have you seen roachpb.Version? https://github.com/cockroachdb/cockroach/blob/77a21a4c711c41a25a5717940e82c06488d5599e/pkg/roachpb/metadata.proto#L403-L419

Perhaps this is the thing to be consumed instead. See the previous comment, cause it's proto-backed, it's versioned and comes with backward compat guarantees. As for this method, aside from making it a pure function, it would be cleaner to write it like the following:
```
func upgradeType(from, to roachpb.Version) string {
    if from.Equal(to) { ... }
    if from.Major != to.Major {
        if from.Less(to) { ... } else { ... }
    }
    
    // Can assume major versions are equal, so do the same thing for minor versions.
}
```

---

Re: the above, are these operations intended to be used at the level of util.Version, roachpb.Version, or clusterversion.ClusterVersion? (I know, too many versions of versions). If I'm reading the commit message correctly, I suspect what we care about is at the level of clusterversion.ClusterVersion.